portable/NetworkInterface/LPC54018/NetworkInterface.c

```diff
@@ -129,197 +131,205 @@ void ENET_IntCallback( ENET_Type * base,
-                            pxBufferDescriptor->xDataLength = length;
-
-                            if( eConsiderFrameForProcessing( pxBufferDescriptor->pucEthernetBuffer ) == eProcessBuffer )
-                            {
-                                xRxEvent.eEventType = eNetworkRxEvent;
-                                xRxEvent.pvData = ( void * ) pxBufferDescriptor;
-
-                                if( xSendEventStructToIPTask( &xRxEvent, 0 ) == pdFALSE )
-                                {
-                                    vReleaseNetworkBufferAndDescriptor( pxBufferDescriptor );
-                                    iptraceETHERNET_RX_EVENT_LOST();
-                                    PRINTF( "RX Event Lost\n" );
-                                }
-                                else
-                                {
-                                    /* Message successfully transfered to the stack */
-                                }
-                            }
-                            else
-                            {
-                                PRINTF( "RX Event not to be considered\n" );
-                                vReleaseNetworkBufferAndDescriptor( pxBufferDescriptor );
-                                /* Not sure if a trace is required.  The stack did not want this message */
-                            }
-                        }
-                        else
-                        {
-                            PRINTF( "RX No Buffer Available\n" );
-                            ENET_ReadFrame( ENET, &g_handle, NULL, 0, 0 );
-                            /* No buffer available to receive this message */
-                            iptraceFAILED_TO_OBTAIN_NETWORK_BUFFER();
-                        }
-                    }
-
-                    break;
-
-                case kStatus_ENET_RxFrameEmpty: /* Received an empty frame.  Ignore it */
-                    receiving = pdFALSE;
-                    break;
-
-                case kStatus_ENET_RxFrameError: /* Received an error frame.  Read & drop it */
-                    PRINTF( "RX Receive Error\n" );
-                    ENET_ReadFrame( ENET, &g_handle, NULL, 0, 0 );
-                    /* Not sure if a trace is required.  The MAC had an error and needed to dump bytes */
-                    break;
-
-                default:
-                    PRINTF( "RX Receive default\n" );
-                    break;
-            }
-        }
-    }
+uint32_t length;
+NetworkBufferDescriptor_t *pxBufferDescriptor;
+IPStackEvent_t xRxEvent;
+status_t status;
+
+	while( pdTRUE )
+	{
+		BaseType_t notified = ulTaskNotifyTake( pdTRUE, pdMS_TO_TICKS(500) );
+
+		if(notified == pdFALSE) // no RX packets for a bit so check for a link
+		{
+			PHY_GetLinkStatus( &phyHandle, &g_linkStatus );
+		}
+		else // got a packet notification so process it.
+		{
+			BaseType_t receiving = pdTRUE;
+
+			while( ( receiving == pdTRUE ) && notified )
```

Comment:
Is the check on 'notified' necessary? If the control reached at the while, notified must be `pdTRUE`.
I did not find any instance of notified being modified inside the loop too.

Reply:
Yes, I noticed the same: `notified` will not change during that loop.

Reply:
Ooops, left in an earlier change. Thanks for noticing.